### ISSUE TITLE
CNMN.Image.crunch efficiency concerns

### ISSUE THREAD

**Issue by bclindner** [bclindner/CNMN#9]
Current image crunch pipeline is decent, but reliance on Mogrify (& the underlying ImageMagick CLI) seriously limits efficiency. Another transformation method would be preferred.

See [Wikipedia article on seam carving](https://en.wikipedia.org/wiki/Seam_carving) - this is a well-known algorithm, and there are several ways I can think of to go about this:

- Direct use of ImageMagick API via NIF
- Direct use of liblqr via NIF
  - Ideally it would be nice to put out a bounty or donate code to remove glibc as this is the only blocker for #8 
- Write the seam carving algorithm directly into Elixir/Erlang (probably via something GPU-able like [axon](https://hexdocs.pm/axon/Axon.html) or [nx](https://hex.pm/packages/nx))
  - [instructions for implementation](https://web.archive.org/web/20220921061042/https://www.statfe.com/projects/csc5280_project4/) (albeit link rotted) that would provide a basis for this
  - This looks mathy but potentially very doable for Elixir

Basically any of this work should probably go in its own separately published Hex library unless it is extremely brief to write
